packages/nimble-components/src/table/tests/table-keyboard-navigation.spec.ts

```diff
@@ -0,0 +1,1109 @@
+
+                    await verifyLastTabKeyEventBehavior(keyEvent);
+                });
+
+                it('clicking a cell in a new row, then a cell in the original row, will focus the clicked (original) row', async () => {
+                    await clickAndSendFocusEvents(pageObject.getCell(1, 0));
+                    await clickAndSendFocusEvents(pageObject.getCell(0, 0));
+
+                    expect(currentFocusedElement()).toBe(pageObject.getRow(0));
+                });
+            });
+
+            it('DownArrow key press will focus the cell in the same column in the next row', async () => {
+                await sendKeyPressesToTable(keyArrowDown, keyArrowRight); // cell 0, 1
+
+                await sendKeyPressToTable(keyArrowDown);
+                expect(currentFocusedElement()).toBe(pageObject.getCell(1, 1));
+            });
+
+            it('UpArrow key press will focus the cell in the same column in the previous row', async () => {
+                await sendKeyPressesToTable(
+                    keyArrowDown,
+                    keyArrowDown,
+                    keyArrowRight
+                ); // cell 1, 1
+
+                await sendKeyPressToTable(keyArrowUp);
+                expect(currentFocusedElement()).toBe(pageObject.getCell(0, 1));
+            });
+
+            it('a row is tabindex 0 when focused, and tabindex -1 when clicked away from', async () => {
+                await sendKeyPressesToTable(keyArrowDown, keyArrowLeft); // focus row
+                const firstRow = pageObject.getRow(0);
+
+                expect(firstRow.tabIndex).toBe(0);
+
+                await clickAndSendFocusEvents(pageObject.getCell(1, 0));
+
+                expect(firstRow.tabIndex).toBe(-1);
+            });
+
+            it('when a row is focused, all action menus in that row are visible (and otherwise hidden) #SkipFirefox #SkipWebkit', async () => {
+                await addActionMenu(column2);
+                await addActionMenu(column3);
+                await sendKeyPressesToTable(keyArrowDown, keyArrowLeft); // focus row
+
+                expect(pageObject.isCellActionMenuVisible(0, 1)).toBe(true);
+                expect(pageObject.isCellActionMenuVisible(0, 2)).toBe(true);
+
+                await sendKeyPressToTable(keyArrowRight); // focus cell 0, 0
+                expect(pageObject.isCellActionMenuVisible(0, 1)).toBe(false);
+                expect(pageObject.isCellActionMenuVisible(0, 2)).toBe(false);
+            });
+
+            it('when a cell is focused, its action menu is visible #SkipFirefox #SkipWebkit', async () => {
+                await addActionMenu(column1);
+
+                await sendKeyPressToTable(keyArrowDown);
+
+                expect(pageObject.isCellActionMenuVisible(0, 0)).toBe(true);
+            });
+
+            it('when a cell with an action menu is focused, pressing Ctrl-Enter will open the action menu', async () => {
```

Comment:
Thoughts on adding an assert / similar test that arrow keys on the action menu cause its menu items to be selected? On the one hand that's testing action menu behavior so maybe not appropriate here. But on the other hand I could imagine the table's key handling getting in the way of the menu's key handling so at least a sanity check seems prudent.

Reply:
I hit some issues when fully interacting with action menus from these tests - basically once the menu was open, the active element wasn't a menu item, it was `body`.

I added tests to cover using UpArrow/DownArrow to open the action menu, which at least partially covers that (since once the action menu button is focused, the table itself isn't handling arrow key presses, so I can check that the focused cell is the same afterwards).